stdlib/2and3/pickle.pyi

```diff
@@ -1,20 +1,46 @@
 import sys
-from typing import Any, IO, Mapping, Union, Tuple, Callable, Optional, Iterator
+from typing import Any, IO, Mapping, Union, Tuple, Callable, Optional, Iterable, Iterator
 
 HIGHEST_PROTOCOL: int
 if sys.version_info >= (3, 0):
     DEFAULT_PROTOCOL: int
 
 
-if sys.version_info >= (3, 0):
+if sys.version_info >= (3, 8):
+    # TODO: holistic design for buffer interface (typing.Buffer?)
+
+    class PickleBuffer:
+        # buffer must be a buffer-providing object
+        def __init__(self, buffer: Any): ...
+        def raw(self) -> memoryview: ...
+        def release(self) -> None: ...
+    def dump(
+        obj: Any, file: IO[bytes], protocol: Optional[int] = ..., *,
+        fix_imports: bool = ...,
+        buffer_callback: Optional[Callable[[PickleBuffer], None]] = ...
```

Comment:
If I understand the documentation correctly, the callback can return a truish value to signal in-band data. I think we should just use `Any` as return type for the callback. Also affects other methods below. I'd also suggest to define a type alias for the callback type (e.g. `_BufferCallback`) and use it throughout the file.